rplugin/python3/deoplete/sources/deoplete_jedi/worker.py

```diff
@@ -62,16 +62,12 @@ def completion_work(self, cache_key, extra_modules, source, line, col,
     def run(self):
         while True:
             try:
-                work = self.in_queue.get(block=False, timeout=0.5)
+                work = self.in_queue.get()
                 self.log.debug('Got work')
-                self.out_queue.put(self.completion_work(*work), block=False)
+                self.out_queue.put(self.completion_work(*work))
```

Comment:
Here I had the timeout left, since it sounded like having a `Full` exception thrown would indicate some problem and might be useful in that regard, e.g. the server needs to be restarted then.

From the doc:

>         If optional args 'block' is true and 'timeout' is None (the default),
        block if necessary until a free slot is available. If 'timeout' is
        a non-negative number, it blocks at most 'timeout' seconds and raises
        the Full exception if no free slot was available within that time.
        Otherwise ('block' is false), put an item on the queue if a free slot
        is immediately available, else raise the Full exception ('timeout'
        is ignored in that case).